src/coreclr/vm/jitinterfacegen.cpp

```diff
@@ -104,6 +105,7 @@ void InitJITHelpers1()
             SetJitHelperFunction(CORINFO_HELP_NEWARR_1_OBJ, JIT_NewArr1OBJ_UP);
 
             ECall::DynamicallyAssignFCallImpl(GetEEFuncEntryPoint(AllocateStringFastUP), ECall::FastAllocateString);
+            ECall::DynamicallyAssignFCallImpl(GetEEFuncEntryPoint(JIT_BoxFastUP), ECall::NonNullableBox);
```

Comment:
This block is Windows specific, so this change only helps on Windows. We need an equivalent fix for non-Windows platforms too.

Reply:
Yeah I'll probably revert this particular change and make an equivalent one that's more xplat-friendly.

Reply:
I've inlined the trail allocation logic into JIT_Box and factored out the rest of the logic to be in a separate framed helper instead of using the dynamic FCALL impl logic.

Let me know if this looks good to you.